lib/kubernetes-deploy/kubernetes_resource/deployment.rb

```diff
@@ -30,12 +31,24 @@ def sync
             )
             pod.deploy_started = @deploy_started
             pod.interpret_json_data(pod_json)
+
+            if !@representative_pod && pod_probably_new?(pod_json)
```

Comment:
This makes me 😢  / 😳 , but it lets us provide the huge benefit of pod logs/events on deployments without waiting for much bigger (200+ LOC) changes required to do this properly. That code _is already written_ so ideally the hack will be in the wild for a very short time. I think this is worth doing to get the output changes in front of users sooner. Risks: 
- If there is massive clock skew between the cluster and the deployer, no representative pod will be chosen, and we'll show the "go look in Splunk" message anyway.
- If someone manages to deploy twice in 30 seconds and the second deploy fails, we may select the wrong pod.
- If a previous deploy is failing so hard that the pods are being re-created repeatedly (not just restarted, which would be fine) and the new deploy fails too, we may select the wrong pod.